### PR TITLE
SALTO-5729 - salesforce - fix service url for new version of flow upon deploy

### PR DESCRIPTION
**Pull request by rotem531**
fix service url for new version of flow upon deploy

---

service url updates upon deploy for all elements

---
_Release Notes_: 
none

---
_User Notifications_: 
none

### CODE REVIEW

File: packages/salesforce-adapter/src/metadata_deploy.ts [hunk 1]
```diff
@@ -606,12 +609,22 @@ export const deployMetadata = async (
...
     const changeElem = getChangeData(change)
     const changeDeployedIds = changeToDeployedIds[changeElem.elemID.getFullName()]
     // TODO - this logic is not perfect, it might produce false positives when there are
     // child xml instances (because we pass in everything with a single change)
-    return successfulFullNames.some(successfulId => changeDeployedIds[successfulId.type]?.has(successfulId.fullName))
+    const metadataId = successfulFullNames.find(successfulId =>
+      changeDeployedIds[successfulId.type]?.has(successfulId.fullName),
+    )
+    if (metadataId) {
+      if (fetchProfile.isFeatureEnabled('shouldPopulateInternalIdAfterDeploy')) {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/metadata_deploy.ts:622]:
```suggestion
    if (metadataId && fetchProfile.isFeatureEnabled('shouldPopulateInternalIdAfterDeploy')) {
```
Avoid nested ifs when possible

File: packages/salesforce-adapter/test/adapter.crud.test.ts
Comment (tamtamirr) [inline packages/salesforce-adapter/test/adapter.crud.test.ts:1]: There's many diffs on this file 🤔 Can we avoid that? I'd expect having only insertions of the new test cases.

File: packages/salesforce-adapter/test/filters/elements_url.test.ts
Comment (tamtamirr) [inline packages/salesforce-adapter/test/filters/elements_url.test.ts:1]: There should be dedicated test cases for `onFetch` and `onDeploy`. Let's discuss that F2F 🥹

**Review by tamtamirr — COMMENTED** [review]
Great work 🐊  Added some comments.
This PR is also missing release notes.

File: packages/salesforce-adapter/src/types.ts [hunk 2]
```diff
@@ -104,7 +104,9 @@ const OPTIONAL_FEATURES = [
   'genAiReferences',
   'networkReferences',
   'extendFetchTargets',
+  'shouldPopulateInternalIdAfterDeploy',
   'addParentToInstancesWithinFolder',
+  'shouldPopulateInternalIdAfterDeploy',
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/types.ts:109]: unique this

File: packages/salesforce-adapter/test/adapter.crud.test.ts [hunk 3]
```diff
@@ -1417,298 +1486,344 @@ describe('SalesforceAdapter CRUD', () => {
...
-            ...mockDefaultValues.Profile,
-            description: 'Updated profile description',
-          },
-          mockTypes.Profile,
-        )
-      })
-
-      describe('when the request succeeds', () => {
-        const DEPLOYMENT_ID = 'testDeploymentId'
+      describe('for non profile instances', () => {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/adapter.crud.test.ts:1489]:
```suggestion
      describe('when the internal ID of the Element changes', () => {
```
We shouldn't specify implementation detail in the `describe` and `it`. We should describe the scenario we are testing and not the fact we invoke the code on non Profile instance. (This is simply an edge-case in the adapter that is not possible in Salesforce)

File: packages/salesforce-adapter/test/adapter.crud.test.ts [hunk 4]
```diff
@@ -1417,298 +1486,344 @@ describe('SalesforceAdapter CRUD', () => {
...
             progressReporter: nullProgressReporter,
           })
         })
-        it('should return an InstanceElement', () => {
-          expect(result.appliedChanges).toHaveLength(1)
-          expect(getChangeData(result.appliedChanges[0])).toBeInstanceOf(InstanceElement)
+        it('should update internal id for non profile instances', async () => {
+          const a = _.get(getChangeData(result.appliedChanges[0]), 'value')[constants.INTERNAL_ID_FIELD]
+          expect(a).not.toBe('ApexPageId')
+          expect(a).toBe('new_id')
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/adapter.crud.test.ts:1530]:
```suggestion
          const deployedInstance = getChangeData(result.appliedChanges[0]) as InstanceElement
          expect(deployedInstance).toBeInstanceOf(InstanceElement)
          expect(deloyedInstance.value[constants.INTERNAL_ID_FIELD]).toEqual('new_id')
```
What is `a` 🤔 ?
Consider the following suggestion

File: packages/salesforce-adapter/test/adapter.crud.test.ts [hunk 5]
```diff
@@ -1417,298 +1486,344 @@ describe('SalesforceAdapter CRUD', () => {
...
             mockDeployResult({
               id: DEPLOYMENT_ID,
               success: true,
               componentSuccess: [
                 {
-                  fullName: mockDefaultValues.Profile.fullName,
-                  componentType: constants.PROFILE_METADATA_TYPE,
+                  fullName: mockDefaultValues.ApexPage.fullName,
+                  componentType: constants.APEX_PAGE_METADATA_TYPE,
+                  id: 'new_id',
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/adapter.crud.test.ts:1508]: Lets const `new_id` and use it in the test as-well.

File: packages/salesforce-adapter/test/filters/elements_url.test.ts [hunk 6]
```diff
@@ -40,100 +41,137 @@ describe('elements url filter', () => {
...
+        expect(res.errors).toHaveLength(1)
+        expect(err[0]).toEqual({
+          severity: 'Warning',
+          message: WARNING_MESSAGE,
+          detailedMessage: WARNING_MESSAGE,
+        })
+      })
+    })
+  })
+  describe('onDeploy tests', () => {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/filters/elements_url.test.ts:157]:
```suggestion
  describe('onDeploy', () => {
```

File: packages/salesforce-adapter/test/filters/elements_url.test.ts [hunk 7]
```diff
@@ -40,100 +41,137 @@ describe('elements url filter', () => {
       annotations: { apiName: 'Account', metadataType: 'CustomObject' },
     })
   })
+  describe('onFetch tests', () => {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/filters/elements_url.test.ts:44]:
```suggestion
  describe('onFetch', () => {
```

File: packages/salesforce-adapter/test/filters/elements_url.test.ts [hunk 8]
```diff
@@ -40,100 +41,137 @@ describe('elements url filter', () => {
...
+      const elementsUrlRetrieverSpy = jest.spyOn(ElementsUrlRetrieverModule, 'lightningElementsUrlRetriever')
+
+      beforeEach(() => {
+        elementsUrlRetrieverSpy.mockImplementation(() => {
+          throw new Error()
+        })
+      })
+
+      afterEach(() => {
+        elementsUrlRetrieverSpy.mockRestore()
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/filters/elements_url.test.ts:134]: Why this change? 🤔

**Review by tamtamirr — COMMENTED** [review]
Great work! Added some comments

**Review by tamtamirr — APPROVED** [review]
Great work 👑
